ctapipe/io/simteleventsource.py

```diff
@@ -513,15 +511,14 @@ def __init__(self, input_url=Undefined, config=None, parent=None, **kwargs):
         )
         if self.back_seekable and self.is_stream:
             raise IOError("back seekable was required but not possible for inputfile")
-
-        self._subarray_info = self.prepare_subarray_info(
-            self.file_.telescope_descriptions, self.file_.header
-        )
```

Comment:
Any particular reason for this move?

Reply:
Yes, it needs to come after parse_simulation_header, otherwise I can't use the SimulationConfigurationContainer